### PR TITLE
#186 Integrate hyperscript-helpers into cycle dom

### PR DESCRIPTION
**Pull request by Cmdv**
Steps taken towards completion of adding hyperscript-helpers to cycle-dom
- [x] Implement hyperscript-helper
- [x] Convert existing tests
- [x] Create simple new test to make sure hyperscript-helpers is being pulled in?
- [x] Update docs

### CODE REVIEW

File: src/cycle-dom.js [hunk 1]
```diff
@@ -3,6 +3,17 @@ let {makeDOMDriver} = require(`./render-dom`)
...
+      bdi, bdo, blockquote, body, br, button, canvas, caption,
+      cite, code, col, colgroup, dd, del, dfn, dir, div, dl,
+      dt, em, embed, fieldset, figcaption, figure, footer, form,
+      h1, h2, h3, h4, h5, h6, head, header, hgroup, hr, html,
+      i, iframe, img, input, ins, kbd, keygen, label, legend,
+      li, link, map, mark, menu, meta, nav, noscript, object,
+      ol, optgroup, option, p, param, pre, q, rp, rt, ruby, s,
+      samp, script, section, select, small, source, span, strong,
+      style, sub, sup, table, tbody, td, textarea, tfoot, th,
+      thead, title, tr, u, ul, video} = require(`hyperscript-helpers`)(h)
```
Comment (staltz) [inline src/cycle-dom.js]: Replace this with 
```js
let hh = require(`hyperscript-helpers`)(h)
```

File: src/cycle-dom.js [hunk 2]
```diff
@@ -62,6 +73,25 @@ let CycleDOM = {
...
+  bdi, bdo, blockquote, body, br, button, canvas, caption,
+  cite, code, col, colgroup, dd, del, dfn, dir, div, dl,
+  dt, em, embed, fieldset, figcaption, figure, footer, form,
+  h1, h2, h3, h4, h5, h6, head, header, hgroup, hr, html,
+  i, iframe, img, input, ins, kbd, keygen, label, legend,
+  li, link, map, mark, menu, meta, nav, noscript, object,
+  ol, optgroup, option, p, param, pre, q, rp, rt, ruby, s,
+  samp, script, section, select, small, source, span, strong,
+  style, sub, sup, table, tbody, td, textarea, tfoot, th,
+  thead, title, tr, u, ul, video,
```
Comment (staltz) [inline src/cycle-dom.js]: Replace this with `...hh`

File: test/browser/custom-elements.js [hunk 3]
```diff
@@ -67,7 +67,7 @@ describe('Custom Elements', function () {
       return {
         DOM: Rx.Observable.combineLatest(ext.props.get('color'), number$,
           function (color, number) {
-            return h('h3.stateful-element', {style: {color}}, String(number));
+            return h3('.stateful-element', {style: {color}}, String(number));
```
Comment (staltz) [inline test/browser/custom-elements.js]: :+1:

**Comment by Cmdv** [comment]
ah cool yeah I was wondering about a exporting everything in a better format!!
sweet I'll make the changes,

Do you think we should do a single test for `hh` just feels like all the other test cover that really?

**Comment by staltz** [comment]
No need to make more tests...

**Comment by Cmdv** [comment]
Yeah thats what I thought, cool I'll make changes and push up when I get the time.

File: src/cycle-dom.js [hunk 4]
```diff
@@ -62,6 +63,15 @@ let CycleDOM = {
   h,
 
   /**
+   * shortcuts to [hyperscript-helpers](
+   * https://github.com/ohanhi/hyperscript-helpers).
+   * This is a helper for extending virtual-hyperscript. Create virtual DOM elements with `div({className: 'wrapper'}, [ h1('Header') ])` instead of `h('div.wrapper', [ h('h1', 'Header') ])`.
+   * @name hh
+   */
+
+  hh,
```
Comment (staltz) [inline src/cycle-dom.js]: This should be `...hh` not `hh`. But don't worry, I'll merge it and continue from here.

**Comment by Frikki** [comment]
@staltz `...hh` is ES2016 proposal. Neither babel 5 nor babel 6 transpiles object spread correctly. The current code in auto-scope doesn’t transpile correctly. Check the */lib*.

File: src/cycle-dom.js (hunk 4 above)
Comment (Cmdv) [inline src/cycle-dom.js]: ahhh, darn it! but @Frikki + @TylorS spotted that things might have accidentally been right in the first place!! :-S

**Comment by staltz** [comment]
@Frikki looks fine to me:

```js
var CycleDOM = _extends({
  makeDOMDriver: makeDOMDriver,
  makeHTMLDriver: makeHTMLDriver,
  h: h
}, hh, {
  hJSX: hJSX,
  svg: svg,
  // ...
});
```

**Comment by Frikki** [comment]
@staltz You’re correct and I was mistaken. I thought the spread operator would create `div, a, h3, ... //etc`.

**Pull request merged**